### PR TITLE
feat: Use text field in the location property

### PR DESCRIPTION
**Pull request by zucchinidev**
### Checklist:

* [x] Have you added Draft Release Notes in `docs/draft-release-notes.md`?
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

[#182299591](https://www.pivotaltracker.com/story/show/182299591)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,18 +5,21 @@ go 1.18
 require (
 	code.cloudfoundry.org/jsonry v1.1.3
 	github.com/Pallinder/go-randomdata v1.2.0
-	github.com/cloudfoundry/cloud-service-broker v0.10.1
```
Comment (blgm) [inline go.mod:8]: Typically we don't update this.

**Review by pivotal-marcela-campo — CHANGES_REQUESTED** [review]

File: staticcheck.conf [hunk 2]
```diff
@@ -0,0 +1,2 @@
+checks = ["all"]
+dot_import_whitelist=["github.com/onsi/gomega", "github.com/onsi/gomega/gexec", "github.com/onsi/ginkgo/v2"]
```
Comment (blgm) [inline staticcheck.conf:2]: 1) Why are we still using Ginkgo v1?
2) I thought that staticcheck would not complain about dot imports in test files, so I'm surprised it was needed. We don't do it in the main Cloud Service Broker repo: https://github.com/cloudfoundry/cloud-service-broker/blob/538380d5f2df45b3a104758c8ec2de9a7304ae22/staticcheck.conf#L1

**Review by blgm — CHANGES_REQUESTED** [review]
Mostly looks good as far as I can tell. The hard thing about this is that there are many features in this PR:
- changing region from enum to string
- adding `go fmt` and `staticcheck` linting and fixing the issues
- adding integration tests
- adding a GitHub action

How hard would it be to split this PR up into individual features? If it's too hard, then I'm happy to merge it like this. In general a PR should contain only one feature. I'm ok with some incidental additional changes, but in this case it's hard to reason about the changes independently.

File: go.mod [hunk 3]
```diff
@@ -5,18 +5,21 @@ go 1.18
...
+	github.com/cloudfoundry/cloud-service-broker v0.10.2-0.20220622110201-538380d5f2df
```
[the remark this answers is not in the document]
Comment (zucchinidev) [reply go.mod:8]: Thank you. I made a mistake!!

File: go.mod (hunk 1 above)
Comment (zucchinidev) [reply go.mod:8]: Thank you very much, @blgm. I made a mistake adding this modification. I'll change it now.

As for splitting this PR. It sounds good to me! I'll start now. I'm sorry I mixed up concepts

File: staticcheck.conf (hunk 2 above)
Comment (zucchinidev) [reply staticcheck.conf:2]: Hey @blgm, thank you very much for reviewing the PR. 
In the case of the static check rule, unfortunately we are using gomega in our acceptance-test that are not test files.
image

As for your first question, I don't understand exactly what you mean. We are using v2. Do you want to have a short meeting?

**Pull request closed without merging**